g4root/src/TG4RootDetectorConstruction.cxx

```diff
@@ -374,37 +371,79 @@ G4Material* TG4RootDetectorConstruction::CreateG4Material(
     pMaterial =
       new G4Material(name, density, nComponents, state, temp, pressure);
     for (Int_t i = 0; i < nComponents; i++) {
-      //         TGeoElement *elem = mixt->GetElement(i);
-      //         name = elem->GetTitle();
-      //         G4Element *pElement = G4Element::GetElement(name);
-      TGeoElement* elem = table->GetElement(Int_t(mixt->GetZmixt()[i]));
-      if (!elem) {
-        G4ExceptionDescription description;
-        description << "      "
-                    << "Woops: no element corresponding to Z="
-                    << Int_t(mixt->GetZmixt()[i]);
-        G4Exception("TG4RootDetectorConstruction::CreateG4Material",
-          "G4Root_F006", FatalException, description);
-      }
-      elname = elem->GetTitle();
-      symbol = elem->GetName();
-      G4Element* pElement =
-        new G4Element(elname, symbol, G4double(mixt->GetZmixt()[i]),
-          G4double(mixt->GetAmixt()[i]) * (g / mole));
+      TGeoElement* elem = mixt->GetElement(i);
+      G4Element* pElement = CreateG4Element(elem);
       pMaterial->AddElement(pElement, mixt->GetWmixt()[i]);
     }
   }
   else {
     // Materials with 1 element.
     //      G4cout << "Creating G4 material "<< name << G4endl;
-    pMaterial = new G4Material(name, G4double(mat->GetZ()),
-      mat->GetA() * g / mole, density, state, temp, pressure);
+    if (mat->GetElement()->GetNisotopes() >
+        0) { // user-defined material with isotopes
+      G4Element* pElement = CreateG4Element(mat->GetElement());
+      pMaterial = new G4Material(name, density, 1, state, temp, pressure);
+      pMaterial->AddElement(pElement, 1.);
+    }
+    else { // standard NIST element
+      pMaterial = new G4Material(name, G4double(mat->GetZ()),
+        mat->GetA() * g / mole, density, state, temp, pressure);
+    }
   }
+
   fG4MaterialMap.insert(G4MaterialVal_t(mat, pMaterial));
   //   G4cout << pMaterial << G4endl;
   return pMaterial;
 }
 
+//______________________________________________________________________________
+G4Element* TG4RootDetectorConstruction::CreateG4Element(TGeoElement* elem)
+{
+  G4Element* pElement;
+
+  // if such element is already created:
+  if (pElement =
+        G4Element::GetElement(elem->GetTitle(), /* G4bool warning = */ false)) {
```

Comment:
This code produces a compilation warning:
```
.../TG4RootDetectorConstruction.cxx:405:16: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
  if (pElement =
      ~~~~~~~~~^
.../TG4RootDetectorConstruction.cxx:405:16: note: place parentheses around the assignment to silence this warning
  if (pElement =
               ^
      (
.../TG4RootDetectorConstruction.cxx:405:16: note: use '==' to turn this assignment into an equality comparison
  if (pElement =
               ^
               ==
```

Please, change it as follows, and **do not squash commits**, as this makes reviewing of updates more difficult. The commits will be squashed when PR is merged.
```cpp
  G4Element* pElement =
    G4Element::GetElement(elem->GetTitle(), /* G4bool warning = */ false);

  // if such element is already created
  if (pElement != nullptr) {
    return pElement;
  }
```

Reply:
Done!